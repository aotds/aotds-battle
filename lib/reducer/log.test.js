"use strict";

var _log = require("./log");

var _log2 = _interopRequireDefault(_log);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

test('logs', () => {
  var _state;

  let state = (0, _log2.default)(undefined, {
    type: 'STUFF',
    meta: {
      id: 123
    }
  });
  state = (0, _log2.default)(state, {
    type: 'FOO',
    meta: {
      id: 124,
      parent_action_id: 123
    }
  });
  state = (0, _log2.default)(state, {
    type: 'BAR',
    meta: {
      id: 125
    }
  });
  expect(state).toMatchSnapshot('state');
  expect((_state = state, (0, _log.tree_log)(_state))).toMatchSnapshot('tree');
});