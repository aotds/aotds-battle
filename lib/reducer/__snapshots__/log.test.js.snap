// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs: state 1`] = `
Array [
  Object {
    "meta": Object {
      "id": 123,
    },
    "type": "STUFF",
  },
  Object {
    "meta": Object {
      "id": 124,
      "parent_action_id": 123,
    },
    "type": "FOO",
  },
  Object {
    "meta": Object {
      "id": 125,
    },
    "type": "BAR",
  },
]
`;

exports[`logs: tree 1`] = `
Array [
  Object {
    "meta": Object {
      "child_actions": Array [
        Object {
          "meta": Object {
            "id": 124,
            "parent_action_id": 123,
          },
          "type": "FOO",
        },
      ],
      "id": 123,
    },
    "type": "STUFF",
  },
  Object {
    "meta": Object {
      "id": 125,
    },
    "type": "BAR",
  },
]
`;
